### PR TITLE
(GH-69) Extend documentation

### LINKED ISSUE #69: Extend documentation for "Optional method arguments"
At the moment the documentation [Examples](https://bbtsoftware.github.io/BBT.Maybe/docs/usage/examples#optional-method-arguments)  feature only a very basic description.
We should add the example from the issue #65 to the docs, shouldn't be a great deal as the code is already there.

### PR DESCRIPTION
**Pull request by marco-bertschi**
Fixes #69

### CODE REVIEW

File: docs/input/docs/usage/examples.md [hunk 1]
```diff
@@ -55,6 +55,25 @@ public class Data()
 }
 ```
 
+If a default value is needed, for example when specifying a file extension, the value can be encapsulated by a `Maybe` as well:
+
+```csharp
+/// Data class with mandatory property Foo and optional property MaybeBar. Default of MaybeBar is Maybe.None<Bar>.
```
Comment (pascalberger) [inline docs/input/docs/usage/examples.md]: Something seems to be wrong here. There's neither a `MaybeBar` in the example nor a `Bar` type

File: docs/input/docs/usage/examples.md [hunk 2]
```diff
@@ -67,6 +86,19 @@ public void Foo(Maybe<Bar> maybeBar = default)
 }
 ```
 
+It's also possible to retrieve the value (or a default value) from the `Maybe` in a method:
+
+```csharp
+private readonly defaultBar = new Bar();
+
+/// maybeBar is an optional parameter. Maybe.None<Bar>() is default,
```
Comment (pascalberger) [inline docs/input/docs/usage/examples.md]: `Maybe.None<Bar>() is the default value.`

File: docs/input/docs/usage/examples.md [hunk 3]
```diff
@@ -67,6 +86,19 @@ public void Foo(Maybe<Bar> maybeBar = default)
...
+/// but using ValueOrDefault a default instance of the bBar object can be used.
```
Comment (pascalberger) [inline docs/input/docs/usage/examples.md]: `By using ValueOrDefault a default instance of the Bar class can be used.`

**Review by pascalberger — CHANGES_REQUESTED** [review]
Some smaller issues

File: docs/input/docs/usage/examples.md (hunk 1 above)
Comment (marco-bertschi) [reply docs/input/docs/usage/examples.md]: Thx, fixed it.

File: docs/input/docs/usage/examples.md (hunk 3 above)
Comment (marco-bertschi) [reply docs/input/docs/usage/examples.md]: Thx, fixed it.

File: docs/input/docs/usage/examples.md (hunk 2 above)
Comment (marco-bertschi) [reply docs/input/docs/usage/examples.md]: Thx, fixed it.

**Review by pascalberger — APPROVED** [review]
LGTM

**Pull request merged**